db/crud.go

```diff
@@ -1107,13 +1140,13 @@ func (db *Database) documentUpdateFunc(docExists bool, doc *Document, allowImpor
 		return
 	}
 
-	// FIXME: This stored body stuff can be changed with the upcoming revcache work
+	//// FIXME: This stored body stuff can be changed with the upcoming revcache work
 	body := newDoc.Body()
-	storedBody := body
-	if newDoc.Deleted {
-		storedBody[BodyDeleted] = true
-	}
-	storedBody = stripSpecialProperties(storedBody)
+	//storedBody := body
```

Comment:
Can we add doc.GetMutableBody that unmarshals BodyBytes when present, or does a deep copy of Body if not present?  This will avoid the deep copy below when we're working with BodyBytes.